x-pack/plugins/ml/public/components/job_selector/job_selector_table/job_selector_table.js

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+
+import React, { Fragment, useState, useEffect } from 'react';
+import { PropTypes } from 'prop-types';
+import { CustomSelectionTable } from '../custom_selection_table';
+import { getBadge } from '../job_selector';
+import { TimeRangeBar } from '../timerange_bar/';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLoadingSpinner,
+  EuiTabbedContent,
+} from '@elastic/eui';
+
+import {
+  LEFT_ALIGNMENT,
+  CENTER_ALIGNMENT,
+  SortableProperties,
+} from '@elastic/eui/lib/services';
+import { i18n } from '@kbn/i18n';
+
+
+const JOB_FILTER_FIELDS = ['job_id', 'groups'];
+const GROUP_FILTER_FIELDS = ['id'];
+
+export function JobSelectorTable({
+  ganttBarWidth,
+  groupsList,
+  jobs,
+  onSelection,
+  selectedIds,
+  singleSelection,
+  timeseriesOnly
+}) {
+  const [sortableProperties, setSortableProperties] = useState();
+  const [currentTab, setCurrentTab] = useState('Jobs');
+
+  useEffect(() => {
+    let sortablePropertyItems = [];
+    let defaultSortProperty = 'job_id';
+
+    if (currentTab === 'Jobs' || singleSelection) {
+      sortablePropertyItems = [
+        {
+          name: 'job_id',
+          getValue: item => item.job_id.toLowerCase(),
+          isAscending: true,
+        },
+        {
+          name: 'groups',
+          getValue: item => (item.groups ? item.groups[0].toLowerCase() : ''),
+          isAscending: true,
+        }
+      ];
+    } else if (currentTab === 'Groups') {
+      defaultSortProperty = 'id';
+      sortablePropertyItems = [
+        {
+          name: 'id',
+          getValue: item => item.id.toLowerCase(),
+          isAscending: true,
+        }
+      ];
+    }
+    const sortableProps = new SortableProperties(sortablePropertyItems, defaultSortProperty);
+
+    setSortableProperties(sortableProps);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [jobs, currentTab]);
+
+  const tabs = [{
+    id: 'Jobs',
+    name: i18n.translate('xpack.ml.jobSelector.jobsTab', {
+      defaultMessage: 'Jobs',
+    }),
+    content: renderJobsTable(),
+  },
+  {
+    id: 'Groups',
+    name: i18n.translate('xpack.ml.jobSelector.groupsTab', {
+      defaultMessage: 'Groups',
+    }),
+    content: renderGroupsTable()
+  }];
+
+  function getGroupOptions() {
```

Comment:
Nice to have: Could be a `<GroupOptions />` stateless functional component.

Reply:
This is used to create `field_value_selection` for Eui's `SearchBar` component so it is expecting a particular format.